src/components/Activity.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import {STATIC_DATABASE_URL} from '../config.json';
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+import { FaBicycle, FaRunning, FaWalking, FaClock, FaMapMarkerAlt, FaLeaf, FaPiggyBank, FaMap } from 'react-icons/fa';
+
+const Activity = () => {
+    const [activities, setActivities] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    const activityIcons = {
+        cycling: <FaBicycle className="text-blue-500 text-4xl" />,
+        running: <FaRunning className="text-red-500 text-4xl" />,
+        walking: <FaWalking className="text-green-500 text-4xl" />,
+    };
+  
+    useEffect(() => {
+      const fetchActivities = async () => {
+        try {
+          const response = await axios.get(
+            `${STATIC_DATABASE_URL}/activities`,
+          );
+          if (response.status != 200) {
+            throw new Error('Network response was not ok');
+          }
+          setActivities(response.data); 
+          setLoading(false);
+        } catch (error) {
+          setError(error);
+          setLoading(false);
+        }
+      };
+  
+      fetchActivities();
+    }, []);
+
+    if (loading) {
+        return <div className="max-w-4xl mx-auto p-4">Loading...</div>;
+    }
+
+    if (error) {
+        return <div className="max-w-4xl mx-auto p-4">Error: {error.message}</div>;
+    }
+
+    const { id } = useParams();
```

Comment:
React complains when `useParams()` is called after the conditional returns of lines 40 and 44. Could you move this call to after the other hooks on lines 9-11?

```
Activity.jsx:47 Warning: React has detected a change in the order of Hooks called by Activity. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useState                   useState
2. useState                   useState
3. useState                   useState
4. useEffect                  useEffect
5. undefined                  useContext
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ```